### PR TITLE
feat(TDKN-238): Allow unauthenticated access to Prometheus

### PR DESCRIPTION
**Pull request by fhuaulme**
**What is the problem this Pull Request is trying to solve?**

See https://jira.talendforge.org/browse/TDKN-238 (description part)
 
**What is the chosen solution to this problem?**
 
See https://jira.talendforge.org/browse/TDKN-238 (proposal part)

Feature
* Add `talend.security.allowPublicPrometheusEndpoint` to allow unauthenticated access to Prometheus endpoint (dev mode only, see Jira description).
* Documentation update.
* Example update.

Misc
* (improvement) use `WebEndpointProperties` to propertly secure actuator endpoints in case service configuration changes endpoint base path.

**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDKN-238
 
**Please check if the Pull Request fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/25/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/25/)

File: daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/token/TokenSecurityConfiguration.java [hunk 1]
```diff
@@ -60,10 +76,27 @@ public TokenSecurityConfiguration(@Value("${talend.security.token.value:}") Stri
...
-        registry = registry.requestMatchers(toAnyEndpoint()).hasRole(TokenAuthentication.ROLE);
-        for (String protectedPath : ADDITIONAL_PROTECTED_PATHS) {
+        for (String protectedPath : PROTECTED_PATHS) {
             registry = registry.antMatchers(protectedPath).hasRole(TokenAuthentication.ROLE);
         }
+        // Configure actuator
+        for (PathMappedEndpoint actuatorEndpoint : actuatorEndpoints) {
+            final String rootPath = actuatorEndpoint.getRootPath();
+            final boolean enforceTokenUsage;
+            if (allowPrometheusUnauthenticatedAccess && "prometheus".equals(rootPath)) {
```
Comment (tlnd-mtallet) [inline daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/token/TokenSecurityConfiguration.java]: This condition will not work in the current kubernetes setup for example.
For each spring-boot service deployed in kubernetes that want to be automatically integrated with the prometheus collector we have this in the service configuration
```yaml
management:
  server:
    port: 9002
  endpoints:
    web:
      base-path: /
      exposure:
        include: health,info,prometheus,bindings
      path-mapping:
        prometheus: metrics
```
In the current setup of the operator without any custom configuration we have to override the path for the `prometheus` endpoint. So the equals on the root path is nice but it'll not work for most of talend services.

So my proposal here would be to use their `EndpointId` abstraction and something like 
```java
if (allowPrometheusUnauthenticatedAccess && actuatorEndpoint.equals(actuatorEndpoints.get(EndpointId.of("prometheus"))) {
...
}
```

Comment (fhuaulme) [reply daikon-spring/daikon-spring-security/src/main/java/org/talend/daikon/security/token/TokenSecurityConfiguration.java]: Yes, good point 👍

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/28/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/28/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/29/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/29/)

**Pull request merged**